docs/modules/tiered-storage/pages/overview.adoc

```diff
@@ -35,3 +39,26 @@ Tiered Storage uses the following:
 - Data Persistence and Tiered Storage are mutually exclusive features. If both are enabled simultaneously, `InvalidConfigurationException` is thrown during the member startup.
 
 - SQL is not supported for the Tiered-Storage-backed maps. `UnsupportedOperationException` is thrown when SQL query is executed. To avoid this, use xref:query:predicate-overview.adoc[Predicate API] instead.
+
+== Supported Devices
+
+Hazelcast supports only NVMe (local) SSD devices for Tiered Storage.
+While HDDs and some network storages work with the same file abstraction that local SSDs do, and technically speaking such setups work, the access latency of these devices compared to that of the modern SSDs are so high that using Tiered Storage with such devices are not supported.
+
+== High-Level Design Overview
+
+The Tiered Storage feature for IMaps is backed by a special-purpose, addressable data structure called HybridLog.
+This data structure spans over the memory and the configured disk making the aggregated capacity of the resources available for storing IMap's entries.
+There is a HybridLog data structure for every partition of every IMap.
+
+The IMap entries stored in these HybridLog instances are internally referenced by their logical addresses.
+The key to logical address mapping is done by a hash index component.
+These hash indices are hashtables using separate chaining for conflict resolution.
+The table of these hash indices are stored in the native memory, whereas the chaining information of the buckets are stored on the disk.
+
+Tiered Storage makes sure every accessed IMap entry is available in the memory to build a hot cache of entries.
+This means, if an entry is on the disk, first it gets relocated to the memory inside the HybridLog hosting the entry, and its previous location on the disk becomes unreachable.
+As more and more entries get unreachable, to prevent running out of disk space the disk space occupied by these unreachable entries has to be reclaimed.
+This reclamation is done by a compactor running synchronously with regards to the IMap operations of the same partition.
+
+The design of Tiered Storage is based on the concepts and data structures introduced by the https://www.microsoft.com/en-us/research/uploads/prod/2018/03/faster-sigmod18.pdf[FASTER: A Concurrent Key-Value Store with In-Place Updates] Microsoft research paper.
```

Comment:
How open do we want to be about the internal implementation of TS in this doc?

Reply:
Good question. We can avoid naming the structure and removing the reference to the paper. But if we ever want to write a blog post about the internals, it's inevitable to mention FASTER (or we will be criticized for not giving credit to its authors).

It's a valid question though if we want our manual referring to external links.

Reply:
We try not to reference papers directly in the product documentation, instead we are referencing to our blogs in case the reader wants to know more in-depth information.

Reply:
Removed the reference to FASTER in 06a2a152059bfb69e1201dfdbf7842c9b65f589f